src/h5web/visualizations/line/LineVis.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect } from 'react';
+import { extent } from 'd3-array';
+import shallow from 'zustand/shallow';
+import styles from './LineVis.module.css';
+import DataLine from './DataLine';
+import VisCanvas from '../shared/VisCanvas';
+import { Domain, AxisOffsets } from '../shared/models';
+import PanZoomMesh from '../shared/PanZoomMesh';
+import { extendDomain } from '../shared/utils';
+import { useLineVisConfig } from './config';
+
+interface Props {
+  data: number[];
+  axisOffsets: AxisOffsets;
+}
+
+function LineVis(props: Props): JSX.Element {
+  const { data, axisOffsets } = props;
+  const [setAbscissaDomain, setOrdinateDomain] = useLineVisConfig(
+    state => [state.setAbscissaDomain, state.setOrdinateDomain],
+    shallow
+  );
+
+  const xDomain = extendDomain([0, data.length - 1], 0.1);
+  const extentD3 = extent(data);
+  const yDomain: Domain | undefined =
+    extentD3[0] !== undefined && extentD3[1] !== undefined
+      ? extendDomain(extentD3, 0.1)
+      : undefined;
```

Comment:
I extend both x and y domains by 10% to avoid clipping of the curve ends by the axes. This can be tuned/made more clever later.